lib/ex_aws/s3/impl.ex

```diff
@@ -232,6 +234,14 @@ defmodule ExAws.S3.Impl do
     body_binary = body |> IO.iodata_to_binary
 
     request(client, :post, bucket, "/?delete", body: body_binary, headers: %{"content-md5" => content_md5})
+    |> case do
+      {:ok, body} -> 
+        case tail == [] do
+          true -> {:ok, body}
+          false -> delete_multiple_objects(client, bucket, tail, opts)
```

Comment:
regarding this, would we really want to pass back a list of results where each result is ok? I was thinking rather that the happy path result should be either {:ok, _} or {:ok, number_deleted}

When a failure occurs we of course have to return the failure reason. I guess we could return the number successfully deleted, as well as the error for the failure... I am a little wary of returning the list of items deleted and the error as returned data load gets large. But what do you think?